c++/src/kj/compat/http.c++

```diff
@@ -2446,35 +2453,28 @@ public:
 
     amount = kj::min(amount, length);
     length -= amount;
+    uint64_t actual = amount;
 
-    auto promise = amount == 0
-        ? kj::Promise<uint64_t>(amount)
-        : getInner().pumpBodyFrom(input, amount).then([this,amount](uint64_t actual) {
-      // Adjust for bytes not written.
+    if (amount != 0) {
+      actual = co_await getInner().pumpBodyFrom(input, amount);
       length += amount - actual;
       if (length == 0) doneWriting();
-      return actual;
-    });
+    }
 
     if (overshot) {
-      promise = promise.then([amount,&input](uint64_t actual) -> kj::Promise<uint64_t> {
         if (actual == amount) {
           // We read exactly the amount expected. In order to detect an overshoot, we have to
```

Comment:
Not for this PR, but it appears there's a pre-existing missing optimization: Earlier in the function, if `tryGetLength()` returned non-null and we verified the length was correct there, then we don't actually have to try reading this extra byte here...